app_dart/lib/src/request_handlers/push_gold_status_to_github.dart

```diff
@@ -261,24 +259,57 @@ class PushGoldStatusToGithub extends ApiRequestHandler<Body> {
   ) async {
     final Stream<IssueComment> comments = gitHubClient.issues.listCommentsByIssue(slug, pr.number);
     await for (IssueComment comment in comments) {
-      if (comment.body.contains('Changes reported for pull request #${pr.number} at sha ${pr.head.sha}')) {
+      if (comment.body.contains(config.flutterGoldCommentID(pr))) {
         return true;
       }
     }
     return false;
   }
+}
 
-  Future<bool> _isFirstComment(
-    GitHub gitHubClient,
-    PullRequest pr,
-    RepositorySlug slug,
-  ) async {
-    final Stream<IssueComment> comments = gitHubClient.issues.listCommentsByIssue(slug, pr.number);
-    await for (IssueComment comment in comments) {
-      if (comment.body.contains('Golden file changes have been found for this pull request')) {
-        return false;
-      }
+Future<Map<String, dynamic>> _queryGraphQL(Logging log, GraphQLClient client, int prNumber) async {
+  final QueryResult result = await client.query(
+    QueryOptions(
+      document: pullRequestChecksQuery,
+      fetchPolicy: FetchPolicy.noCache,
+      variables: <String, dynamic>{
+        'sPullRequest': prNumber,
+      },
+    ),
+  );
+
+  if (result.hasErrors) {
+    for (GraphQLError error in result.errors) {
+      log.error(error.toString());
     }
-    return true;
+    throw const BadRequestException('GraphQL query failed');
   }
+  return result.data as Map<String, dynamic>;
 }
+
+const String pullRequestChecksQuery = r'''
+query ChecksForPullRequest($sPullRequest: int!) {
+  repository(owner: "flutter", name: "flutter") {
+    pullRequest(number: $sPullRequest) {
+      commits(last: 1) {
+        nodes {
+          commit {
+            # (appId: 64368) == flutter-dashboard. We only care about
+            # flutter-dashboard checks.
+            checkSuites(last: 1, filterBy: {appId: 64368}) {
```

Comment:
I have not been able to run this in the github graphql explorer. I based it off of the check for waiting pull requests pattern. How can I check this?

Reply:
Unfortunately there is no way to test it with the explorer as the data from preview apis is not available. What I did was to create a temporary handler  call it with post and print the result from the query.